adm_program/modules/preferences/preferences.php

```diff
@@ -513,25 +513,38 @@ function getPreferencePanel($group, $id, $parentId, $title, $icon, $body)
     $formValues['mail_sendmail_name'],
     array('maxLength' => 50, 'helpTextIdInline' => 'SYS_SENDER_NAME_DESC')
 );
-$selectBoxEntries = array(0 => $gL10n->get('SYS_HIDDEN'), 1 => $gL10n->get('SYS_SENDER'), 2 => $gL10n->get('SYS_ADMINISTRATOR'));
+$selectBoxEntries = array(0 => $gL10n->get('SYS_MAIL_BULK'), 1 => $gL10n->get('SYS_MAIL_SINGLE'));
 $formEmailDispatch->addSelectBox(
-    'mail_recipients_with_roles',
-    $gL10n->get('SYS_MULTIPLE_RECIPIENTS'),
+    'mail_sending_mode',
+    $gL10n->get('SYS_MAIL_SENDING_MODE'),
     $selectBoxEntries,
-    array('defaultValue' => $formValues['mail_recipients_with_roles'], 'showContextDependentFirstEntry' => false, 'helpTextIdInline' => 'SYS_MULTIPLE_RECIPIENTS_DESC')
-);
-$formEmailDispatch->addCheckbox(
-    'mail_into_to',
-    $gL10n->get('SYS_INTO_TO'),
-    (bool) $formValues['mail_into_to'],
-    array('helpTextIdInline' => 'SYS_INTO_TO_DESC')
-);
-$formEmailDispatch->addInput(
-    'mail_number_recipients',
-    $gL10n->get('SYS_NUMBER_RECIPIENTS'),
-    $formValues['mail_number_recipients'],
-    array('type' => 'number', 'minNumber' => 0, 'maxNumber' => 9999, 'step' => 1, 'helpTextIdInline' => 'SYS_NUMBER_RECIPIENTS_DESC')
+    array('defaultValue' => $formValues['mail_sending_mode'], 'showContextDependentFirstEntry' => false, 'helpTextIdInline' => 'SYS_MAIL_SENDING_MODE_DESC')
 );
+
+$sendingMode = $gSettingsManager->getInt('mail_sending_mode');
+
+// Only show this options if SendingMode = 0 / SYS_MAIL_BULK
+if($sendingMode == 0) {
```

Comment:
This wont work in the form because you check the mode within php. If you want to show or hide those fields this must be done with javascript dynamically if the user select each mode. Otherwise the prefences should be shown always.